src/sagemaker/tensorflow/sm_training.rst

```diff
@@ -0,0 +1,147 @@
+* ``SM_OUTPUT_DATA_DIR``: A string that represents the path to the directory to write output artifacts to.
+  Output artifacts might include checkpoints, graphs, and other files to save, but do not include model artifacts.
+  These artifacts are compressed and uploaded to S3 to an S3 bucket with the same prefix as the model artifacts.
+* ``SM_CHANNEL_XXXX``: A string that represents the path to the directory that contains the input data for the specified channel.
+  For example, if you specify two input channels in the MXNet estimator's ``fit`` call, named 'train' and 'test', the environment variables ``SM_CHANNEL_TRAIN`` and ``SM_CHANNEL_TEST`` are set.
+
+For the exhaustive list of available environment variables, see the `SageMaker Containers documentation <https://github.com/aws/sagemaker-containers#list-of-provided-environment-variables-by-sagemaker-containers>`__.
+
+A typical training script loads data from the input channels, configures training with hyperparameters, trains a model, and saves a model to ``model_dir`` so that it can be deployed for inference later.
+Hyperparameters are passed to your script as arguments and can be retrieved with an ``argparse.ArgumentParser`` instance.
+For example, a training script might start with the following:
+
+.. code:: python
+
+    import argparse
+    import os
+
+    if __name__ =='__main__':
+
+        parser = argparse.ArgumentParser()
+
+        # hyperparameters sent by the client are passed as command-line arguments to the script.
+        parser.add_argument('--epochs', type=int, default=10)
+        parser.add_argument('--batch-size', type=int, default=100)
+        parser.add_argument('--learning-rate', type=float, default=0.1)
+
+        # input data and model directories
+        parser.add_argument('--model_dir', type=str)
+        parser.add_argument('--sm-model-dir', type=str, default=os.environ['SM_MODEL_DIR'])
+        parser.add_argument('--train', type=str, default=os.environ['SM_CHANNEL_TRAIN'])
+        parser.add_argument('--test', type=str, default=os.environ['SM_CHANNEL_TEST'])
+
+        args, _ = parser.parse_known_args()
+
+        # ... load from args.train and args.test, train a model, write model to args.model_dir.
+
+Because the SageMaker imports your training script, you should put your training code in a main guard (``if __name__=='__main__':``) if you are using the same script to host your model,
+so that SageMaker does not inadvertently run your training code at the wrong point in execution.
+
+Note that SageMaker doesn't support argparse actions.
+If you want to use, for example, boolean hyperparameters, you need to specify ``type`` as ``bool`` in your script and provide an explicit ``True`` or ``False`` value for this hyperparameter when instantiating your TensorFlow estimator.
+
+For more on training environment variables, please visit `SageMaker Containers <https://github.com/aws/sagemaker-containers>`_.
+
+Adapt your TensorFlow script
```

Comment:
s/Adapt/Adapting

this header is a bit vague - are you talking about adapting from a legacy mode script or adapting from a random example?

Reply:
random example. I will try to make this clearer.